samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,215 @@
+
+This introductory sample shows how to use ML.NET to predict the relevance and order of hotel search results. In the world of machine learning, this type of prediction is known as ranking.
+
+## Problem
+The ability to perform ranking is a common problem faced by search engines since users expect query results to be ranked\sorted according to their relevance.  This problem extends beyond the needs of search engines to include a variety of business scenarios where personalized sorting is key to the user experience.  Here are a few specific examples:
+* Travel Agency - Provide a list of hotels with those that are most likely to be purchased\booked by the user positioned highest in the list.
+* Shopping - Display items from a product catalog in an order that aligns with a user's shopping preferences.
+* Recruiting - Retrieve job applications ranked according to the candidates that are most qualified for a new job opening.
+
+Ranking is useful to any scenario where it is important to list items in an order that increases the likelihood of a click, purchase, reservation, etc.
+ 
+In this sample, we show how to apply ranking to the first example listed above to rank hotel search results according to the likelihood that the hotel will be purchased\booked by the user. To perform ranking, there are two algorithms currently available - FastTree Boosting (FastRank) and Light Gradient Boosting (LightGbm).  We use the LightGbm's Lambdarank algorithm in this sample to automatically build an ML model to predict ranking. 
+
+## Dataset
+The training and testing data used by this sample is based on a public [dataset available at Kaggle](https://www.kaggle.com/c/expedia-personalized-sort) originally provided by Expedia (https://www.expedia.com).
+
+Expedia's datasets consist of hotel search results that are grouped according to a user's query; each hotel result includes the following details:
+* Hotel attributes, such as location attractiveness and price.
+* User's criteria for searching hotels, such as the number of rooms\children\adults, length of stay, etc.
+* User's purchase and browsing history, such as whether they clicked the link of a hotel or purchased\booked it.
+* Information on similar competitor hotel offerings.
+
+## ML Task - Ranking
+As previously mentioned, this sample uses the LightGbm Lambdarank algorithm which is applied using a supervised learning technique known as "Learning to Rank".  This technique requires that train/test datasets contain groups of data instances that are labeled with their ideal ranking value.  The label is a numerical\ordinal value, such as {4, 3, 2, 1, 0} or a text value {"Perfect", "Excellent", "Good", "Fair", or "Bad"}.  The process for labeling these data instances with their ideal ranking value can be done manually by subject matter experts.  Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. This sample uses the latter approach.
```

Comment:
Has someone checked if `{"Perfect", "Excellent", "Good", "Fair", or "Bad"}` still work as Labels? This works from the MAML command-line, but I'm uncertain that the estimators API accepts anything besides {4, 3, 2, 1, 0}.

Reply:
I haven't tried it (I lifted this point from the TLC documentation and assumed it was still applicable).  I will verify it to confirm.

Reply:
I followed up on this by changing my label values to strings ("Perfect", "Fair", "Bad") - when I attempt to train the model I get an Invalid Op exception saying the following, so it looks like this isn't supported.  Does a bug need to be logged for this?

"Splitter/consolidator worker encountered exception while consuming source data.
"Could not parse value Bad in line 62, column Label".

Reply:
Based on my above comment, I have removed the mention of using strings as label values. #resolved